FlowCal/mef.py

```diff
@@ -149,30 +168,66 @@ def selection_std(populations,
     low, high : int or float
         Low and high thresholds. Required if the elements in `populations`
         are numpy arrays. If not specified, and the elements in
-        `populations` are FCSData objects, use 0.015 of the lowest value
-        and 0.0985 of the highest value in ``populations[0].range``.
+        `populations` are FCSData objects, use 1.5% and 98.5% of the range
+        in ``populations[0].range``.
     n_std_low, n_std_high : float, optional
         Number of standard deviations from `low` and `high`, respectively,
         that a population's mean has to be closer than to be discarded.
+    scale : str, optional
+        Rescaling applied to `populations` before calculating means and
+        standard deviations. If ``linear``, no rescaling is applied. If
+        ``log``, apply base-10 logarithm to `populations`.
 
     Returns
     -------
     selected_mask : boolean array
         Flags indicating whether a population has been selected.
 
     """
+    # Check that ``scale`` is supported
+    if scale!='linear' and scale!='log':
+        raise ValueError("scale {} not supported".format(scale))
+
     # Default thresholds
     if low is None:
-        if hasattr(populations[0], 'range'):
-            low = 0.015*populations[0].range(0)[0]
+        if hasattr(populations[0], 'hist_bins'):
+            if scale=='linear':
+                r = populations[0].range(0)
+                low = r[0] + 0.015*(r[1] - r[0])
+            elif scale=='log':
+                r = populations[0].hist_bins(channels=0, nbins=1, scale='log')
+                low = 10**(np.log10(r[0]) + 0.015*(np.log10(r[1]/r[0])))
         else:
             raise TypeError("argument 'low' not specified")
     if high is None:
-        if hasattr(populations[0], 'range'):
-            high = 0.985*populations[0].range(0)[-1]
+        if hasattr(populations[0], 'hist_bins'):
+            if scale=='linear':
+                r = populations[0].range(0)
+                low = r[0] + 0.985*(r[1] - r[0])
+            elif scale=='log':
+                r = populations[0].hist_bins(channels=0, nbins=1, scale='log')
+                high = 10**(np.log10(r[0]) + 0.985*(np.log10(r[1]/r[0])))
         else:
             raise TypeError("argument 'high' not specified")
 
+    # Rescale if necessary
+    if scale=='log':
+        # Copy events
+        for i in range(len(populations)):
+            populations[i] = populations[i].copy()
+
+        # Log scale requires saturating negative and zero-valued events
+        for p in populations:
+            p[p < 1] = 1.
```

Comment:
not sure what you mean by this comment. (shouldn't they already be saturated by the instrument? we're manually doing the same thing here?). Also, is `p<1` the right check? not sure what kind of data we have at this point, but with floating point, I imagine you could have valid (i.e. non-saturating) events between 0 and 1. Seems like the right check should be `p<=0`, with violators being replaced with something positive (which opens a new can of worms regarding what positive value should be used)

you sure it's not more appropriate to just discard values less than or equal to 0 when `scale=log`? instead of artificially designating them as `1.`

(seems like this change won't affect the behavior of this function much, though, so this might now matter much)